server/src/main/java/org/elasticsearch/action/admin/cluster/stats/DenseVectorFieldStats.java

```diff
@@ -9,19 +9,23 @@
 
 package org.elasticsearch.action.admin.cluster.stats;
 
-import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.xcontent.XContentBuilder;
 
 import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Objects;
 
 /**
  * Holds enhanced stats about a dense vector mapped field.
  */
 public final class DenseVectorFieldStats extends FieldStats {
     static final int UNSET = -1;
-
+    static final String NOT_INDEXED = "not_indexed";
+    Map<String, Integer> vectorIndexTypeCount; // count of mappings by index type
+    Map<String, Integer> vectorSimilarityTypeCount; // count of mappings by similarity
+    Map<String, Integer> vectorElementTypeCount; // count of mappings by element type
```

Comment:
should we be concerned about these becoming huge (and hence OOMs & co)?

Reply:
@tteofili I wouldn't think so. We have a static number of element types, index types, and similarity types. Since we are counting each separately, we won't have combinatoric explosion. These each will be hashmaps of < 10 elements each or so.